diff -rupN grib_api-1.9.18/configure grib_api-1.9.18_fixed/configure
--- grib_api-1.9.18/configure	2012-10-17 14:58:59.000000000 +0100
+++ grib_api-1.9.18_fixed/configure	2012-12-11 17:28:14.000000000 +0000
@@ -7006,7 +7006,7 @@ $as_echo_n "checking for $compiler optio
     darwin* | rhapsody*)
       # PIC is the default on this platform
       # Common symbols not allowed in MH_DYLIB files
-      lt_prog_compiler_pic='-fno-common'
+      lt_prog_compiler_pic=''
       ;;
 
     hpux*)
@@ -12186,7 +12186,7 @@ $as_echo_n "checking for $compiler optio
     darwin* | rhapsody*)
       # PIC is the default on this platform
       # Common symbols not allowed in MH_DYLIB files
-      lt_prog_compiler_pic_F77='-fno-common'
+      lt_prog_compiler_pic_F77=''
       ;;
 
     hpux*)
@@ -15214,7 +15214,7 @@ $as_echo_n "checking for $compiler optio
     darwin* | rhapsody*)
       # PIC is the default on this platform
       # Common symbols not allowed in MH_DYLIB files
-      lt_prog_compiler_pic_FC='-fno-common'
+      lt_prog_compiler_pic_FC=''
       ;;
 
     hpux*)
