diff -rupN unpack-160114-orig/libmo_unpack/distribute.sh unpack-160114/libmo_unpack/distribute.sh
--- unpack-160114-orig/libmo_unpack/distribute.sh	2014-01-08 11:03:27.000000000 +0000
+++ unpack-160114/libmo_unpack/distribute.sh	2014-05-06 16:53:28.000000000 +0100
@@ -43,8 +43,14 @@ then
   exit 1
 fi
 
-cp -d lib/lib* $synch_dir/lib
-cp -d include/*.h $synch_dir/include
+cp -a lib/lib* $synch_dir/lib
+cp -a include/*.h $synch_dir/include
+
+# Modify install_name on OSX.
+for x in lib/lib*.so*; do
+    install_name_tool -id ${synch_dir%/}/$x ${synch_dir%/}/$x
+done
+
 
 if [ $? -ne 0 ]
 then
diff -rupN unpack-160114-orig/libmo_unpack/make_library unpack-160114/libmo_unpack/make_library
--- unpack-160114-orig/libmo_unpack/make_library	2014-01-08 11:03:27.000000000 +0000
+++ unpack-160114/libmo_unpack/make_library	2014-05-06 16:53:38.000000000 +0100
@@ -59,8 +59,8 @@ then
   has_shared=1
 elif [ "_$os" == "_Darwin" ]
 then
-  CC=gcc
-  LD="ld -shared -soname libmo_unpack.so.$major"
+  CC=cc
+  LD="ld -dylib -undefined dynamic_lookup -install_name @rpath/libmo_unpack.so.$major"
   AR="ar"
   OPTS="-O4 -I include -D_LARGEFILE_SOURCE -D_LARGEFILE_SOURCE64 -D_FILE_OFFSET_BITS=64 -D_DARWIN_SOURCE"
   has_shared=1
